pkgs/build-support/fetchpatch/tests.nix

```diff
@@ -7,7 +7,7 @@ in
 {
   simple = testers.invalidateFetcherByDrvHash fetchpatch {
     url = "https://github.com/facebook/zstd/pull/2724/commits/e1f85dbca3a0ed5ef06c8396912a0914db8dea6a.patch";
-    sha256 = if isFetchpatch2 then "sha256-01BrkHLye4KOdqCw3tv7AJzIF6578pl2fl270TJFTmw=" else "sha256-PuYAqnJWAE+L9bsroOnnBGJhERW8LHrGSLtIEkKU9vg=";
+    sha256 = if isFetchpatch2 then "sha256-w4yU0wt64d0WkuBQPeGf8vn5TH6qSBJvNIgka9QK+/Q=" else "sha256-PuYAqnJWAE+L9bsroOnnBGJhERW8LHrGSLtIEkKU9vg=";
```

Comment:
It's not cached, so I can't check the diff :/ could be https://github.com/NixOS/nixpkgs/issues/257446 as only fetchpatch2 fails while the older version which doesn't include the diff command doesn't fail

![image](https://github.com/NixOS/nixpkgs/assets/56650223/83b80895-2ec2-4aa7-8e0d-334ecb0f1d11)

Reply:
It's not something in Nix or Nixpkgs, because `nix run nix/2.5.1 -- --store ~/nix_2_8 build nixpkgs/41877098f3ac00295a2a9d5e5846201b969a689b#tests.fetchpatch2.simple` behaves the same.

We could add `file://`-based tests in addition to "mirror" the http ones to prevent this kind of situation in the future.

Reply:
Since there is a Nixpkgs issue already, I assume it’s not a blocker for this PR?